doc/languages-frameworks/python.section.md

```diff
@@ -1929,16 +1931,15 @@ Both are also exported in `nix-shell`.
 It is recommended to test packages as part of the build process.
 Source distributions (`sdist`) often include test files, but not always.
 
-By default the command `python setup.py test` is run as part of the
-[`checkPhase`](#ssec-check-phase), but often it is necessary to pass a custom [`checkPhase`](#ssec-check-phase). An
-example of such a situation is when `py.test` is used.
+Previously, by default the command `python setup.py test` is run as part of the
+[`checkPhase`](#ssec-check-phase), but now it is necessary to pass a custom [`checkPhase`](#ssec-check-phase).
+In almost all cases, `pytestCheckHook` is used.
```

Comment:
Were still defaulting to that, aren"t we? Also you don't need to pass a custom `checkPhase`, but rather a hook to set up a certain test framework.

Reply:
It is still `doCheck = true` by default, but the hook (python setup.py test) is not provided unless `format = "setuptools"`. This means that unless we add `pytestCheckHook` to the input or provide checkPhase, test will not run in the current builder.

How about this?
> Previously, by default the command `python setup.py test` is run as part of the [`checkPhase`](#ssec-check-phase), but now it is necessary to pass a test framework hook like `pytestCheckHook`.
It may be necessary to provide a custom checkPhase for packages that do not use a common framework.

Reply:
> By default the `checkPhase` executes the deprecated `python setup.py test` command.

This is the status quo, tests will still try to run `python setup.py test`, when no hook is given. We *could* however change that in a huge treewide change.

> The best practice today is to pass a test hook (e.g. `pytestCheckHook`, `unittestCheckHook`) into `nativeCheckInputs`, that will reconfigure the `checkPhase` to use that particular test framework. Occasionaly it may  be required to provide a custom `checkPhase`, when the package does not rely on a common framework.

Reply:
Do you mean that `python setup.py test` will be executed since `format = "setuptools"` is currently the default?
If so, I agree with you.

Reply:
Oh right, that is depended on the format :facepalm: